tools/check-links

```diff
@@ -137,15 +137,18 @@ class Crawler:
 
     def check_external_links(self, urls):
         with Pool(max_workers=len(urls)) as executor:
-            future_to_url = {executor.submit(check_response, packet.url): packet for packet in urls}
+            future_to_url = {
+                executor.submit(check_response, packet.url): packet for packet in urls
+            }
             for future in future_to_url:
                 packet = future_to_url[future]
                 try:
                     future.result()
                 except UrlNotFound as e:
-                    self.report_error(f"{e.response_code} error", packet)
-                except Exception as exc:
-                    print('%r generated an exception: %s' % (packet.url, exc))
+                    if not requests.get(packet.url).ok:
+                        self.report_error(f"{e.response_code} error", packet)
```

Comment:
Minor: Why is the retry logic here and not next to where the request is made (line 62)? This seems like an odd place to retry.